src/PathSimulator.cpp

```diff
@@ -287,6 +287,66 @@ void PathSimulator<DDPackage>::generateAlternatingSimulationPath(std::size_t sta
     setSimulationPath(components, true);
 }
 
+template<class DDPackage>
+void PathSimulator<DDPackage>::generateGatecostSimulationPath(std::size_t startingPoint, std::list<std::size_t> gateCosts) {
+    typename SimulationPath::Components components{};
+    components.reserve(CircuitSimulator<DDPackage>::qc->getNops());
+    std::size_t startElem = startingPoint;
+    components.emplace_back(startElem, startElem + 1);
+    std::size_t leftID   = startElem - 1;
+    std::size_t leftEnd  = 0;
+    std::size_t rightID  = startElem + 2;
+    std::size_t rightEnd = CircuitSimulator<DDPackage>::qc->getNops() + 1;
+    std::size_t nextID   = rightEnd;
+    std::size_t runID    = 0;
+    while (leftID != leftEnd && rightID != rightEnd) {
+        if (runID == 0) {
+            for (auto i = 0U; i < gateCosts.front() - 1; i++) {
+                components.emplace_back(nextID, rightID);
+                nextID++;
```

Comment:
prefer prefix increment (`++nextID`) over postfix increment (`nextID++`) in practically all cases. (please change throughout the whole file and also for the decrement)

Reply:
done and will be included in the latest commit